### ISSUE TITLE
Error updating APIService - resource not found

### ISSUE THREAD

**Issue by halittiryaki** [vadimkim/cert-manager-webhook-hetzner#79]
get a lot of errors in the kube apiserver:

```
E1227 02:11:11.939719       1 controller.go:146] Error updating APIService "v1alpha1.acme.mydomain.tld" with err: failed to download v1alpha1.acme.mydomain.tld: resource not found
E1227 02:11:12.944657       1 controller.go:102] loading OpenAPI spec for "v1alpha1.acme.mydomain.tld" failed with: failed to download v1alpha1.acme.mydomain.tld: resource not found
1227 02:11:12.944687       1 controller.go:109] OpenAPI AggregationController: action for item v1alpha1.acme.mydomain.tld: Rate Limited Requeue.    
```

lots of people have to had it in the past, is this again a compatibility issue?

k8s: v1.30.5
cert.manager: 1.16.2 
hetzner-webhook: 1.31

**Comment by halittiryaki** [comment]
was caused by overriden lib version:
[cert-manager v1.11.1-0.20230119150913-7ebb5f515e4f](https://github.com/vadimkim/cert-manager-webhook-hetzner/pull/80/commits/6fe21a3f8a7dc0450b519b9bc9398faa3f6a2acf#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L9)

however, the PR also updates other dependencies.
until the PR is merged or the fixes are released otherwise, this image can be used:
[dockerhub-halittiryaki/cert-manager-webhook-hetzner/1.3.0](https://hub.docker.com/layers/halittiryaki/cert-manager-webhook-hetzner/1.3.0/images/sha256-ca8f6368ff9b381611b17465593588577bccd9f0dcf44c1ebe8af57d388cff45)

**Comment by josipnedic** [comment]
I got exactly the same problem, error is logged every 2 minutes.

Looking forward to the next release which includes the updated dependencies.

I'm using:

- k3s: `v1.31.5+k3s1`
- cert-manager: `v1.17.1`
- hetzner-webhook: `1.3.1`

**Comment by vadimkim** [comment]
Try to check with release 1.3.2. Closed unless reported.

**Comment by inola-ww** [comment]
updated chart to 1.3.2, containers not updated - looks like appVersion / image-tag stays at 1.3.0

**Comment by vadimkim** [comment]
You are right, @inola-ww. I pushed it to dockerhub, but chart has github as an image directory... I'll check ghcr.io and push there..

**Comment by vadimkim** [comment]
I pushed new images with tags "latest" and "1.3.2" to ghcr.io. Try it

**Comment by josipnedic** [comment]
I can confirm that the error is gone with version `1.3.2`.

But I had to set the image tag manually since `appVersion` is still set to `1.3.0`:
https://github.com/vadimkim/cert-manager-webhook-hetzner/blob/master/deploy/cert-manager-webhook-hetzner/Chart.yaml#L4

In the `deployment.yaml` the default image tag is retrieved from `appVersion`:
https://github.com/vadimkim/cert-manager-webhook-hetzner/blob/master/deploy/cert-manager-webhook-hetzner/templates/deployment.yaml#L26

You should consider fixing this in the next version and always bump the `helm chart version` and `appVersion` to the same version. This would make upgrading much easier.

Thank you!

**Comment by vadimkim** [comment]
@josipnedic I forgot about appVersion change. Will update Chart and generate new Helm package. I think it will be correct to update existing gh-pages Helm package with version 1.3.2 instead of releasing next version, what do you think?

**Comment by josipnedic** [comment]
Since you already pushed a git tag you should leave `1.3.2` as is.

I would recommend to get straight to `1.3.3` and maybe include some dependency updates or other open issues.

In the worst case `1.3.3` would only fix the `appVersion` but I think this is the cleaner way.

**Comment by vadimkim** [comment]
Tag and binary will not be changed because it is created by "docker buildx ..."  The problem, as you found, is Chart. I.e. current release 1.3.2 references old image with tag "1.3.0".  So I need to update Helm package to reference correct image. I personally think that need to fix 1.3.2 fist before releasing 1.3.3.

**Comment by vadimkim** [comment]
I will update chart and publish Helm package on Monday to keep tradition "no updates on Fridays"
